lib/chef/resource/service.rb

```diff
@@ -175,6 +178,24 @@ def parameters(arg=nil)
         )
       end
 
+      # This will enable user to pass a plist in the case that the filename
```

Comment:
To avoid polluting a single resource with attributes that are unique to many different providers, we've arrived at a dynamic provider resolver system instead of the old single resource -> many providers static map. It's well documented in this blog post: https://www.chef.io/blog/2015/02/10/chef-12-provider-resolver/

In this case, I think you want to create a new ```lib/chef/resource/macosx_service.rb``` resource that inherits from the service resource and adds the additional attributes.